autogpt/memory_management/summary_memory.py

```diff
@@ -7,115 +7,115 @@
 
 cfg = Config()
 
-
-def get_newly_trimmed_messages(
-    full_message_history: List[Dict[str, str]],
-    current_context: List[Dict[str, str]],
-    last_memory_index: int,
-) -> Tuple[List[Dict[str, str]], int]:
+class SummaryMemoryMixin:
+    def get_newly_trimmed_messages(
+        self,
+        current_context: List[Dict[str, str]],
+        last_memory_index: int,
+    ) -> List[Dict[str, str]]:
+        """
+        This function returns a list of dictionaries contained in full_message_history
+        with an index higher than prev_index that are absent from current_context.
+
+        Args:
+            full_message_history (list): A list of dictionaries representing the full message history.
+            current_context (list): A list of dictionaries representing the current context.
+            last_memory_index (int): An integer representing the previous index.
+
+        Returns:
+            list: A list of dictionaries that are in full_message_history with an index higher than last_memory_index and absent from current_context.
+            int: The new index value for use in the next loop.
+        """
+        # Select messages in full_message_history with an index higher than last_memory_index
+        new_messages = [
+            msg for i, msg in enumerate(self.full_message_history) if i > last_memory_index
+        ]
+
+        # Remove messages that are already present in current_context
+        new_messages_not_in_context = [
+            msg for msg in new_messages if msg not in current_context
+        ]
+
+        # Find the index of the last message processed
+        new_index = last_memory_index
+        if new_messages_not_in_context:
+            last_message = new_messages_not_in_context[-1]
+            new_index = self.full_message_history.index(last_message)
+
+        self.last_memory_index = new_index
+
+        return new_messages_not_in_context
```

Comment:
all I did was returning this single value, and then update the 
        self.last_memory_index = new_index